packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,422 @@
+    let request = getRefreshReceiptRequest(properties: properties.isEmpty ? nil : properties)
+    let handler = handlerFactory(request)
+    requestHandlers.insert(handler)
+    handler.startProductRequest { [weak self] response, error in
+      if let error = error {
+        let requestError = FlutterError(
+          code: "storekit_refreshreceiptrequest_platform_error",
+          message: error.localizedDescription,
+          details: error.localizedDescription)
+        completion(requestError)
+        return
+      }
+      completion(nil)
+      self?.requestHandlers.remove(handler)
+    }
+  }
+
+  public func startObservingPaymentQueueWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    getPaymentQueueHandler().startObservingPaymentQueue()
+  }
+
+  public func stopObservingPaymentQueueWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    getPaymentQueueHandler().stopObservingPaymentQueue()
+  }
+
+  public func registerPaymentQueueDelegateWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    #if os(iOS)
+      if #available(iOS 13.0, *) {
+        let messenger = registrar?.messenger()
+        paymentQueueDelegateCallbackChannel = FlutterMethodChannel(
+          name: "plugins.flutter.io/in_app_purchase_payment_queue_delegate",
+          binaryMessenger: messenger!)
+
+        paymentQueueDelegate = FIAPPaymentQueueDelegate(
+          methodChannel: paymentQueueDelegateCallbackChannel!)
+        getPaymentQueueHandler().delegate = (paymentQueueDelegate as! SKPaymentQueueDelegate)
+      }
+    #endif
+  }
+
+  public func removePaymentQueueDelegateWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    #if os(iOS)
+      if #available(iOS 13.0, *) {
+        paymentQueueDelegateCallbackChannel = nil
+        getPaymentQueueHandler().delegate = nil
+        paymentQueueDelegate = nil
+      }
+    #endif
+  }
+
+  public func showPriceConsentIfNeededWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    #if os(iOS)
+      if #available(iOS 13.4, *) {
+        getPaymentQueueHandler().showPriceConsentIfNeeded()
+      }
+    #endif
+  }
+
+  @objc
+  public func handleTransactionsUpdated(_ transactions: [SKPaymentTransaction]) {
+    let maps = transactions.map {
+      FIAObjectTranslator.getMapFrom($0)
+    }
+    transactionObserverCallbackChannel!.invokeMethod("updatedTransactions", arguments: maps)
+  }
+
+  @objc
+  public func handleTransactionsRemoved(_ transactions: [SKPaymentTransaction]) {
+    let maps = transactions.map {
+      FIAObjectTranslator.getMapFrom($0)
+    }
+    transactionObserverCallbackChannel!.invokeMethod("removedTransactions", arguments: maps)
+  }
+
+  @objc
+  public func handleTransactionRestoreFailed(_ error: NSError) {
+    transactionObserverCallbackChannel!.invokeMethod(
```

Comment:
can this be `nil`? can you go over all usage of `!` and make sure either remove it, or document why it's safe to force unwrap it. `!` should be very rarely used, and heavily documented if any.

Reply:
Is using `?` preferable to `!` in most cases? or is it better to unwrap everytime?

Reply:
> or is it better to unwrap everytime

both `?` and `!` are ways of "unwrapping" (since `Optional` type is just an enum wrapper). `!` will crash if it's nil, so we have to be extra careful, and we should avoid it if possible. 

For example, if `transactionObserverCallbackChannel` can be declared as non-optional, do that. Otherwise, add a comment explaining why it has to be an optional, and why we have to force unwrap it here.

Reply:
Okay, I left a comment for transactionObserverCallbackChannel and why it has to be optional